src/nvim/os/input.c

```diff
@@ -159,15 +159,26 @@ bool os_char_avail(void)
 // Check for CTRL-C typed by reading all available characters.
 void os_breakcheck(void)
 {
-  int save_us = updating_screen;
+  static bool recursive = false;
+  const int save_updating_screen = updating_screen;
+
+  // We could be called recursively if stderr is redirected, calling
+  // fill_input_buf() calls settmode() when stdin isn't a tty.  settmode()
+  // calls vgetorpeek() which calls os_breakcheck() again.
```

Comment:
neither `fill_input_buf` nor `settmode` exist in Nvim.  stderr being "redirected" doesn't affect Nvim AFAIK. 

Probably this patch is NA.

Reply:
I've never encountered this problem when building neovim for Debian, which is the context in which I encountered the original problem for Vim.

The problem in Vim happened when stdin is not a tty, causing Vim to open stderr instead.  In Debian's builds, stdin is redirected from `/dev/null` and stderr is usually redirected to a file/pipe.  That combination caused Vim to get cranky.

Reply:
Recursive `os_breakcheck` is essentially a special case of recursive `loop_poll_events` in nvim, which we have decided to forbid by `abort()`. The code we allow to run inside `breakcheck` / `loop_poll_events` is very limited (in what editor state it is allowed to touch). It would be better to unconditionally abort, if this code is accidentally allowed to do something wrong, rather than either `abort()` or either ignore it, depending how the "outer" loop was invoked. (i e: not apply this patch to nvim).